@navikt/core/react/src/overlays/dismissablelayer/DismissableLayer.tsx

```diff
@@ -54,185 +51,131 @@ interface DismissableLayerBaseProps {
-    ref,
+    forwardedRef,
   ) => {
-    const [, setForce] = useState({});
-    const { register, index, descendants } = useDismissableDescendant({
-      disableOutsidePointerEvents,
-      disabled: !enabled,
-      forceUpdate: () => setForce({}),
-    });
-
-    /**
-     * `node` will be set to the ref of the component or nested component
-     * Ex: If
-     * ```
-     * <DismissableLayer asChild>
-     *   <Popover />
-     * </DismissableLayer>
-     * ```
-     * `node` will in this case be the Popover-element.
-     * We use State her and not ref since we want to trigger a rerender when the node changes.
-     */
-    const [node, setNode] = useState<HTMLDivElement | null>(null);
-
-    const mergedRefs = useMergeRefs(setNode, register, ref);
+    const context = useContext(DismissableLayerContext);
 
+    const [, forceRerender] = useState({});
+    const [node, setNode] = React.useState<DismissableLayerElement | null>(
+      null,
+    );
+    const mergedRefs = useMergeRefs(forwardedRef, setNode);
     const ownerDoc = ownerDocument(node);
 
-    const hasInteractedOutsideRef = useRef(false);
-    const hasPointerDownOutsideRef = useRef(false);
-
-    const pointerState = (() => {
-      let lastIndex = -1;
-
-      const descendantNodes = descendants.enabledValues();
-
-      descendantNodes.forEach((obj, _index) => {
-        if (obj.disableOutsidePointerEvents) {
-          lastIndex = _index;
-        }
-      });
-
-      return {
-        /**
-         * Makes sure we stop events at the highest layer with pointer events disabled.
-         * If not checked, we risk closing every layer when clicking outside the layer.
-         */
-        isPointerEventsEnabled: index >= lastIndex,
-        /**
-         * If we find a node with `disableOutsidePointerEvents` we want to disable pointer events on the body.
-         */
-        isBodyPointerEventsDisabled: bodyLockCount > 0,
-        pointerStyle: (index >= lastIndex && bodyLockCount > 0
-          ? "auto"
-          : undefined) as CSSProperties["pointerEvents"] | undefined,
-      };
-    })();
+    /* Layer handling */
+    const layers = getSortedLayers(context.layers, context.branchedLayers);
+    const highestLayerWithOutsidePointerEventsDisabledIndex =
+      findHighestLayerIndex(
+        layers,
+        context.layersWithOutsidePointerEventsDisabled,
+      );
+    const index = node ? layers.indexOf(node) : -1;
+    const isBodyPointerEventsDisabled =
+      context.layersWithOutsidePointerEventsDisabled.size > 0;
+    const shouldEnablePointerEvents =
+      highestLayerWithOutsidePointerEventsDisabledIndex === -1 ||
+      index >= highestLayerWithOutsidePointerEventsDisabledIndex;
 
     /**
      * We want to prevent the Layer from closing when the trigger, anchor element, or its child elements are interacted with.
-     *
      * To achieve this, we check if the event target is the trigger, anchor or a child. If it is, we prevent default event behavior.
```

Comment:
If trigger is just another name for anchor, we should adjust this comment a bit. For example:

```suggestion
    * We want to prevent the Layer from closing when the trigger/anchor element or its child elements are interacted with.
    * To achieve this, we check if the event target is the trigger/anchor or a child. If it is, we prevent default event behavior.
```